activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/cmd/DeployCmd.java

```diff
@@ -126,7 +131,12 @@ protected boolean deploymentsDiffer(DeploymentEntity deployment,
 
         if (deploymentBuilder.hasProjectManifestSet()) {
 
-            return !deployment.getProjectReleaseVersion().equals(saved.getProjectReleaseVersion());
+            if (!deployment.getProjectReleaseVersion().equals(saved.getProjectReleaseVersion())) {
```

Comment:
what about if the project release versions mismatch (returns true)
but the appVersions match instead?

Based on the logic that the enforces appVersion has priority.. would we have a problem in this scenario?

Reply:
The enforced app version has priority only when project release versions mismatch (in the rest of cases is the original upgrade logic that is set into motion, that has been released and fully tested).
This is to make a distinction between the creation of a new pod when we upgrade (and there's a new app version enforced) and the creation of a new pod when there's an outage (no upgrade, no +1 in the counter, so the service relies on the env variable, that if it does not change, effectively means there was no upgrade at all )

Reply:
the enforce app version has priority on everything.
i don't understand why this logic is applied "only when project release versions mismatch"

if you rely on enforced app version first, then you don't have problem with this either:
"This is to make a distinction between the creation of a new pod when we upgrade (and there's a new app version enforced) and the creation of a new pod when there's an outage (no upgrade, no +1 in the counter, so the service relies on the env variable, that if it does not change, effectively means there was no upgrade at all )"